x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/QueryProvider.java

```diff
@@ -107,6 +104,8 @@ public void writeTo(StreamOutput out) throws IOException {
             if (parsingException != null) { // Do we have a parsing error? Throw it
                 if (parsingException instanceof IOException) {
                     throw (IOException) parsingException;
+                } else if (parsingException instanceof RuntimeException) {
+                    throw (RuntimeException) parsingException;
```

Comment:
I added this bit because almost all exceptions used in Elasticsearch are `RuntimeException`s and the bit below that wraps in an `ElasticsearchException` with no message is not great from the point of view of readability of error responses.